modules/nf-core/optitype/tests/main.nf.test

```diff
@@ -0,0 +1,59 @@
+nextflow_process {
+
+    name "Test Process OPTITYPE"
+    script "../main.nf"
+    process "OPTITYPE"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "optitype"
+
+    test("hla - bam") {
+        config "./nextflow.config"
+        when {
+            process {
+                """
+                input[0] = [ [ id:'test', seq_type:'dna' ], // meta map
+                       file(params.test_data['homo_sapiens']['illumina']['test_paired_end_hla_sorted_bam'], checkIfExists: true),
+                       file(params.test_data['homo_sapiens']['illumina']['test_paired_end_hla_sorted_bam_bai'], checkIfExists: true)
+                     ]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(file(process.out.hla_type[0][1]).name).match("bam_hla_type") },
+                { assert snapshot(file(process.out.coverage_plot[0][1]).name).match("bam_coverage_plot") },
+                { assert snapshot(process.out.versions).match("bam_versions") }
+            )
+        }
+    }
+
+    test("hla - bam - cbc") {
+        config "./optitype_cbc.config"
+        when {
+            process {
+                """
+                input[0] = [ [ id:'test', seq_type:'dna' ], // meta map
+                       file(params.test_data['homo_sapiens']['illumina']['test_paired_end_hla_sorted_bam'], checkIfExists: true),
+                       file(params.test_data['homo_sapiens']['illumina']['test_paired_end_hla_sorted_bam_bai'], checkIfExists: true)
+                     ]
+
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(file(process.out.hla_type[0][1]).name).match("bam_cbc_hla_type") },
```

Comment:
Unfortunately the files cannot be simply compared since the objective value (based on the ILP solution) is not 100% stable. But I will figure it out.

Reply:
Not even a header line from the table or something? It would be a little more reliable to check for a string in the file than the file name, but as I said, don't worry if it's not possible.

Reply:
The header line would be possible as well, but I thought it would still be better to check the actual content. I just added a possible solution. For the `cbc` solver we unfortunately get different results for one locus, I guess also because of the low number of reads. I just excluded HLA locus A now for the assertion.

Reply:
Looks like this can happen for both solvers. I will exclude the comparison of HLA-A calls for both tests.